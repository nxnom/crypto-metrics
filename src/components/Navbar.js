import React, { useState } from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import { NavLink, useLocation } from 'react-router-dom';
import { Icon as Iconify } from '@iconify/react';
import { useDispatch, useSelector } from 'react-redux';

import { selectCurrency, setCurrency } from 'src/redux/filter/filterSlice';
import { currencies } from 'src/utils/currency';
import styles from './Navbar.module.css';
import Modal from './Modal';

function Navbar() {
  const [isOpen, setIsOpen] = useState(false);
  const dispatch = useDispatch();
  const currency = useSelector(selectCurrency);
  const { pathname } = useLocation();

  const toggleModal = () => setIsOpen(!isOpen);

  return (
    <header className={styles.header}>
      {pathname === '/' ? (
        <Icon icon="tabler:coin-bitcoin" className={styles.logo} />
      ) : (
        <NavLink to="/" className={styles.back}>
          <Icon icon="material-symbols:arrow-back-ios-new" />
        </NavLink>
      )}
      <h1 className={styles.title}>
        <NavLink to="/">Crypto Metrics</NavLink>
      </h1>
      <button type="button" onClick={toggleModal}>
        <Icon icon="material-symbols:settings" />
      </button>
      <Modal isOpen={isOpen} onClose={toggleModal}>
        <legend className={styles.legend}>Select Currency</legend>
        <div className={styles.radioGroup}>
          {Object.entries(currencies).map(([key, value]) => (
            <div key={key}>
              <input
                type="radio"
                id={key}
                name="currency"
                value={key}
                checked={currency === key}
                onChange={() => {
                  dispatch(setCurrency(key));
                  toggleModal();
                }}
              />
              <label htmlFor={key}>{value.name}</label>
            </div>
          ))}
        </div>
      </Modal>
    </header>
  );
}

function Icon({ icon, className }) {
  return <Iconify className={cn(styles.icon, className)} icon={icon} />;
}

Icon.propTypes = {
  icon: PropTypes.string.isRequired,
  className: PropTypes.string,
};

Icon.defaultProps = {
  className: '',
};

export default Navbar;
