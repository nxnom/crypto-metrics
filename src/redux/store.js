import { configureStore } from '@reduxjs/toolkit';

import api from './services/api';
import filterReducer from './filter/filterSlice';

const store = configureStore({
  reducer: {
    filters: filterReducer,
    [api.reducerPath]: api.reducer,
  },
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat(api.middleware),
});

export default store;
