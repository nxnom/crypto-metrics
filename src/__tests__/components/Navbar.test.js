import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

import Navbar from 'src/components/Navbar';
import store from 'src/redux/store';

describe('Navbar', () => {
  it('renders correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <MemoryRouter>
          <Navbar />
        </MemoryRouter>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
