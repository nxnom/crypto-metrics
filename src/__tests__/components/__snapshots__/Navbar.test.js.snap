// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<div>
  <header
    class="header"
  >
    <span />
    <h1
      class="title"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        Crypto Metrics
      </a>
    </h1>
    <button
      type="button"
    >
      <span />
    </button>
  </header>
</div>
`;
